src/components/PlaceForm/index.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react'
+import FormInput from '../FormInput'
+import queryString from 'query-string'
+import { FormWrapper } from './style.js'
+import './style.js'
+class PlaceForm extends Component {
+    constructor(props) {
+        super(props)
+        const obj = queryString.parse(this.props.history.location.search)
+        this.state = { form: obj }
+    }
+
+    handleChange = name => inputText => {
+        const newform = {
+            ...this.state.form,
+            [name]: inputText,
+        }
+
+        this.setState({ form: newform })
+        this.props.history.push('?' + queryString.stringify(newform))
+    }
+
+    handleSubmit = e => {
+        const { onSubmit, history } = this.props
+        e.preventDefault()
+        onSubmit(this.state.form)
+        this.setState({ form: {} })
+        history.push()
```

Comment:
this will be rewritten as well after you extract this functionality into a HOC React component